### PR TITLE
Port to diode 1.0.0

### PR DESCRIPTION
**Pull request by cquiroz**
This is a small PR updating the diode library to the version 1.0.0. It can be considered pure UI code

### CODE REVIEW

**Comment by jdnavarro** [comment]
Is `ActionType` import necessary now? Everything seems to be based on `Action` now. Am I understanding it correctly?

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/NavBar.scala [hunk 1]
```diff
@@ -35,7 +36,7 @@ object NavBar {
               IconTerminal.copy(p = IconTerminal.p.copy(link = true, circular = true, onClick = Callback {SeqexecCircuit.dispatch(ToggleDevConsole)}))
             )
           ),
-          SeqexecCircuit.connect(_.user)(TopMenu(_))
+          userConnect(x => TopMenu(x))
```
Comment (jdnavarro) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/NavBar.scala]: Perhaps it's the pointfree style bias coming from Haskell but isn't eta reduction a good practice in these cases? Not sure how the syntax would be, would `(TopMenu(_))` work?

**Comment by jdnavarro** [comment]
I'm not sure I understand how diode uses the `ReactConnector` and what are the advantages of separating the connection in 2 steps. Apparently it should save some unnecessary mounting/unmounting but I can't see how it benefits in this case.

**Comment by jdnavarro** [comment]
Overall my comments were mostly nit-picking. With my limited understanding I don't see anything wrong with this PR.

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/NavBar.scala (hunk 1 above)
Comment (cquiroz) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/NavBar.scala]: `(TopMenu(_))` may work, it depends a bit if the compiler can infer the type of x, I'll try

**Comment by cquiroz** [comment]
About `ActionType`. It brings a typeclass into the context to let the calls to `dispatch` accept `Action` making it typesafer

**Comment by cquiroz** [comment]
About `ReactConnector` it is a change on the latest diode, it improves performance by not rebuilding the connector each time the component is re-rendered. Interestingly you can't even compile the old style

For changes check [CHANGELOG](https://github.com/ochrons/diode/blob/master/CHANGES.md)

**Pull request merged**